### PR TITLE
Comparison of Hyperparameter Optimization

### PR DESCRIPTION
**Pull request by leelaylay**
1. ~~Compare different Hyperparameter Optimization algorithms on fake function.~~
2. Compare these algorithms on AutoGBDT example and RocksDB Benchmark.

### CODE REVIEW

File: docs/en_US/Blog/HPOComparison.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    "_type": "randint",
+    "_value": [100]
+  }
+}
+```
+
+The total search space is 1,000,000, the number of maximum trial to 1000. The time limitation is 24 hours.
+
+### Results
+
```
Comment (chicm-ms) [inline docs/en_US/Blog/HPOComparison.md]: 1. Can you explain what is top1 and top5 loss? 
2. Is it necessary to list from top1 to top10 loss?   Considering if you only list top1, top3, top5, top10, do you lose any important information?

File: docs/en_US/Blog/HPOComparison.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
+# Hyperparameter Optimization Comparison
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: can you add more reference link and explain detail about RockDB benchmark?

File: docs/en_US/Blog/HPOComparison.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+*Posted by Anonymous Author*
+
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: Also please add the autoGBDT link

File: docs/en_US/Blog/HPOComparison.md (hunk 1 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: 1. My expression here (top1 and top5 loss) is ambiguous. I mean the top 10 of the least loss by by `Top10 loss`.
2. I don't know is there a better way to list information in multiple experiments. Maybe it's not necessary to do such thing. If we only list top1, top3, top5, top10, we can lose some information.

File: docs/en_US/Blog/HPOComparison.md (hunk 2 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: OK.

File: docs/en_US/Blog/HPOComparison.md (hunk 3 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: I will add it later.

File: docs/en_US/Blog/HPOComparison.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+Comparison of Hyperparameter Optimization algorithms on several problems.
+
+Hyperparameter Optimization algorithms are list below:
+
+- Random Search
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: you could add the link of each tuner.

File: docs/en_US/Blog/HPOComparison.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+- Grid Search
+- Evolution
+- Anneal
+- Metis
+- TPE
+- SMAC
+- HyperBand
+- BOHB
+
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: Could you make a section to tell the detailed environment? Such as the configure of your machine, such as CPU core...

File: docs/en_US/Blog/HPOComparison.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+All algorithms run in NNI local environment.
+
+## First Fake Function
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: First Fake Function? How about “Optimize convex function”?

File: docs/en_US/Blog/HPOComparison.md [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+Fake Function:
+
+![](<http://latex.codecogs.com/gif.latex?Loss(x,y,z)=(x-30+0.1)^2\times(y-40+0.1)^2\times(x+z-50+0.1)^2>)
+
+### Search Space
+
+```json
+{
+  "x": {
+    "_type": "randint",
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: using "randint" means you don't care about the order of value.  I suggested you remove this part.

File: docs/en_US/Blog/HPOComparison.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+Fake Function:
+
+![](http://latex.codecogs.com/gif.latex?Loss(x,y,z)=(x-30+0.1)^2 +(y-40+0.1)^2+(z-50+0.1)^2)
+
+### Search Space
+
+```json
+{
+  "x": {
+    "_type": "randint",
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: same here.

File: docs/en_US/Blog/HPOComparison.md (hunk 6 above)
Comment (xuehui1991) [reply docs/en_US/Blog/HPOComparison.md]: if we remove this part, we don't have to care about the title...

File: docs/en_US/Blog/HPOComparison.md (hunk 1 above)
Comment (chicm-ms) [reply docs/en_US/Blog/HPOComparison.md]: I still do not understand the top10 loss,  is it the average of most smallest 10 loss values?

Comment (chicm-ms) [reply docs/en_US/Blog/HPOComparison.md]: what kind of information is lost if you only list  top1, top3, top5, top10 loss?

Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: Top10 loss I mean here is exact the smallest 10 loss values.

Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: The expression Top10 here is not appropriate and I will consider change it in other words.

File: docs/en_US/Blog/HPOComparison.md (hunk 4 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: OK

File: docs/en_US/Blog/HPOComparison.md (hunk 6 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: I change the title of this section and still keep it.

File: docs/en_US/Blog/HPOComparison.md (hunk 7 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: I still keep it.

File: docs/en_US/Blog/HPOComparison.md (hunk 8 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: I still keep it.

File: docs/en_US/Blog/HPOComparison.md (hunk 1 above)
Comment (chicm-ms) [reply docs/en_US/Blog/HPOComparison.md]: so the 'top10' column is the 10th smallest loss value, and 'top9' column is the 9th smallest loss value?

Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: Yes, that's what I want to express.

File: docs/en_US/Blog/HPOComparison.md (hunk 5 above)
Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: OK.

File: docs/en_US/Blog/HPOComparison.md (hunk 1 above)
Comment (chicm-ms) [reply docs/en_US/Blog/HPOComparison.md]: Ok, as you said the name 'top x loss' here is confusing.
It is not necessary to list all 10 values here,  suggest list only best loss,  average of best 5 losses and average of best 10 losses.

Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: Thank you for your advice.

File: docs/en_US/Blog/HPOComparison.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
+# Hyperparameter Optimization Comparison
+*Posted by Anonymous Author*
+
+Comparison of Hyperparameter Optimization algorithms on several problems.
+
```
Comment (xuehui1991) [inline docs/en_US/Blog/HPOComparison.md]: Please add a link from Builtint_Tuner to here.

Comment (leelaylay) [reply docs/en_US/Blog/HPOComparison.md]: OK.

**Pull request merged**